src/main/scala/com/rbmhtechnology/calliope/scaladsl/TransactionalEventProducer.scala

```diff
@@ -148,34 +150,82 @@ class TransactionalEventProducer[A] private(sourceId: String, topic: String, eve
 
   import ProducerGraphRunner._
 
-  private implicit val loggingAdapter = Logging(system, getClass)
+  import scala.concurrent.ExecutionContext.Implicits.global
 
-  private var running = false
+  private implicit val loggingAdapter = Logging(system, getClass)
 
   private val graph = ProducerGraph(
-    EventStoreReader.withTimestampGapDetection(sourceId, eventStore, settings.transactionTimeout),
-    new EventStoreDeleter(eventStore),
+    EventStoreReader.withTimestampGapDetection(sourceId, eventStore, settings.transactionTimeout)(system, ioDispatcher),
+    new EventStoreDeleter(eventStore)(ioDispatcher),
     settings)
 
-  private lazy val graphRunner = system.actorOf(Props(new GraphRunnerSupervisor(graph, onFailure)))
-
-  def run(): EventWriter[A] = {
-    running = true
-    val runner = graphRunner
-    new EventStoreWriter[A](topic, eventStore, runner ! NotifyCommit)
+  private val graphRunner = system.actorOf(Props(new GraphRunnerSupervisor(graph, onFailure)))
+
+  /**
+    * Creates an [[EventWriter]] which is used to store events to the underlying [[EventStore]] and consecutively produce them to the configured
+    * Kafka endpoint.
+    */
+  def eventWriter(): EventWriter[A] =
+    new EventStoreWriter[A](topic, eventStore, graphRunner ! NotifyCommit)
+
+  /**
+    * Starts producing events from the underlying [[EventStore]] to the configured Kafka endpoint.
+    *
+    * This is an asynchronous operation which returns a [[Future]] that is resolved once event production has started or fails if the given
+    * timeout was exceeded.
+    */
+  def run()(implicit timeout: FiniteDuration = 5.seconds): Future[Done] =
```

Comment:
I wonder if we should rather omit the default value here (and below in `stop`) or do we have something similar in other places as well?